distribution/sc-proxy-dex/src/proxy_common.rs

```diff
@@ -31,6 +35,76 @@ pub trait ProxyCommonModule {
         Ok(())
     }
 
+    #[payable("*")]
+    #[endpoint]
+    fn acceptPay(
+        &self,
+        #[payment_token] token_id: TokenIdentifier,
+        #[payment] amount: Self::BigUint,
+    ) {
+        let tx_hash = self.blockchain().get_tx_hash();
+
+        if !self.last_tx_hash().is_empty() {
```

Comment:
I do not know if you have to actually use the txHash. before calling for example exitFarm - you could set the storage of accepted tokens to nil, than you get the call of acceptPay - where you update the storage - and later you verify. It would save some gas. You would use one key for the storage mapper.

Reply:
done